libs/blocks/chart/chart.js

```diff
@@ -178,51 +147,130 @@ const handleIntersect = (chart, chartOptions) => (entries, observer) => {
   });
 };
 
+const getContainerSize = (chartSize, chartType) => {
+  const containerSizes = {
+    area: {
+      small: { height: 345, width: '100%' },
+      medium: { height: 310, width: '100%' },
+      large: { height: 512, width: '100%' },
+    },
+    default: {
+      small: { height: 290, width: '100%' },
+      medium: { height: 295, width: '100%' },
+      large: { height: 350, width: '100%' },
+    },
+    donut: {
+      small: { height: 345, width: '100%' },
+      medium: { height: 450, width: '100%' },
+      large: { height: 512, width: '100%' },
+    },
+    oversizedNumber: {
+      small: { minHeight: 290, width: '100%' },
+      medium: { minHeight: 295, width: '100%' },
+      large: { minHeight: 350, width: '100%' },
+    },
+  };
+  let containerSize = containerSizes?.default?.[chartSize] || {};
+
+  if (chartType in containerSizes) {
+    containerSize = containerSizes?.[chartType]?.[chartSize] || {};
+  }
+
+  return containerSize;
+};
+
+const updateContainerSize = (chartWrapper, chartSize, chartType) => {
+  const containerSize = getContainerSize(chartSize, chartType);
+  const chartRow = chartWrapper?.parentElement;
+
+  if (chartRow) {
+    chartRow.style.width = containerSize?.width;
+    chartRow.style.height = `${containerSize?.height}px`;
+  }
+
+  if (chartWrapper) {
+    chartWrapper.style.width = containerSize?.width;
+    chartWrapper.style.height = `${containerSize?.height}px`;
+  }
+};
+
+const getResponsiveSize = (authoredSize) => {
+  const desktopBreakpoint = 1200;
+  const tabletBreakpoint = 600;
+  const width = window.innerWidth;
+  let size = authoredSize;
+
+  if (width < tabletBreakpoint) {
+    size = SMALL;
+  } else if (width < desktopBreakpoint) {
+    size = MEDIUM;
+  }
+
+  return size;
+};
+
+const handleResize = (el, authoredSize, chartType, data, colors) => {
+  const currentSize = getResponsiveSize(authoredSize);
+  const previousSize = el?.getAttribute('data-responsive-size');
+  const previousIsLarge = previousSize === LARGE;
+  const currentIsLarge = currentSize === LARGE;
+  const chartWrapper = el?.querySelector('.chart_wrapper');
+  const chartInstance = window.echarts?.getInstanceByDom(chartWrapper);
+
+  if (currentSize !== previousSize) {
+    updateContainerSize(chartWrapper, currentSize, chartType);
+    el.setAttribute('data-responsive-size', currentSize);
+  }
+
+  if (previousIsLarge !== currentIsLarge) {
+    chartInstance?.dispose();
+
+    const themeName = getTheme(currentSize);
+    const chart = window.echarts?.init(chartWrapper, themeName, { renderer: 'svg' });
+    const chartOptions = getChartOptions(chartType, data, colors, currentSize);
+
+    chart.setOption(chartOptions);
```

Comment:
Do you need to setup the IntersectionObserver again?

Reply:
I'll look into this and do some testing. If it needs to be updated, I'll create a new PR